lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -1859,6 +1864,47 @@ ChangeResult TypeAnalyzer::visitAtenAddCLikeOp(
   return getLatticeElement(op->getResult(0)).join(knowledge);
 }
 
+ChangeResult TypeAnalyzer::visitAtenNativeBatchNormOp(
+    AtenNativeBatchNormOp op,
+    ArrayRef<LatticeElement<ValueKnowledge> *> operands) {
+  auto input = operands[0]->getValue();
+
+  auto batchNormKnowledge =
+      ValueKnowledge::getNotNonePessimisticValueState(op->getContext());
+  auto meanKnowledge =
+      ValueKnowledge::getNotNonePessimisticValueState(op->getContext());
+  auto invStdKnowledge =
+      ValueKnowledge::getNotNonePessimisticValueState(op->getContext());
+
+  batchNormKnowledge.dtype = input.dtype;
+  meanKnowledge.dtype = input.dtype;
+  invStdKnowledge.dtype = input.dtype;
+
+  // Rank of the input tensor must be greater than or equal to 2. The size of
+  // the input tensor as well as the output tensor should be (N, C, D?, H?, W?).
+  // The running_mean, running_var, weight, and bias should be of size (C).
+  bool training = false;
+  if (matchPattern(op.training(), m_TorchConstantBool(&training)) &&
+      input.hasSizes && input.sizes.size() >= 2) {
+    batchNormKnowledge.hasSizes = true;
+    meanKnowledge.hasSizes = true;
+    invStdKnowledge.hasSizes = true;
+    batchNormKnowledge.sizes = input.sizes;
+    meanKnowledge.sizes.resize(1, 0);
+    invStdKnowledge.sizes.resize(1, 0);
```

Comment:
nit: I think it would be better to directly assign the contents of the `sizes` vectors by doing `...sizes = {0};` This makes it clearer what the vector is and in the case that `sizes` for some reason already contains values, this will ensure that the vector gets set to a single 0 element (`resize(1, 0)` does not fill the value 0 if the original vector is not empty).

Reply:
yeah, {0} makes it more clearer and improves readability. Thanks :)